### PR TITLE
Codejam Canvas

### PR DESCRIPTION
**Pull request by Elaray1**
1. задание - https://github.com/rolling-scopes-school/tasks/blob/master/tasks/stage-2/codejam-canvas/codejam-canvas.md
2. скриншот - https://hostingkartinok.com/show-image.php?id=04ac344e4c9f7c0d81b7a58ca122cd1d
3. 27.10.2019/27.10.2019
4. Maximum points - 50.

+ Layout (menu, tools, canvas) - 10
+ Drawing color matrixes
       4x4 - 10
       32x32 - 10
+ Show image on canvas - 10
+ Dynamic canvas size switcher - 10

Ссылка на хост: https://distracted-mclean-35e83b.netlify.com/
Итоговая оценка: 50 / 50

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/meyer-reset/2.0/reset.css">
+    <link rel="stylesheet" href="style.css">
+  </head>
+  <body>
+
+    <header>
+      <nav>
+        <img src="assets\menu_button.png" alt="menu" width="18" height="12">
+        <h1>CodeJam - Pallete</h1>
```
Comment (karengyuzalyan) [inline index.html]: Wrong title

File: index.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        <ul>
+          <li><img src="assets\current_color.png" alt="" width="20" height="20"><p>Current color</p></li>
+          <li><img src="assets\prev_color.png" alt="" width="20" height="20"><p>Prev color</p></li>
+          <div class="border"></div>
+          <li><img src="assets\red_color.png" alt="" width="20" height="20"><p>red</p></li>
+          <li><img src="assets\blue_color.png" alt="" width="20" height="20"><p>blue</p></li>
+        </ul>
+      </div>
+
```
Comment (karengyuzalyan) [inline index.html]: Redundant emplty lines, please clear

File: script.js [hunk 3]
```diff
@@ -0,0 +1,6269 @@
+window.onload = function() {
+  var canvas = document.getElementById('canvas');
```
Comment (karengyuzalyan) [inline script.js]: As you know what is ES6, please use `const` instaed of `var`

File: script.js [hunk 4]
```diff
@@ -0,0 +1,6269 @@
...
+  ctx = canvas.getContext("2d");
+  var A = [
```
Comment (karengyuzalyan) [inline script.js]: `const`
And also, unclear name of variable, pleaase rename

File: script.js [hunk 5]
```diff
@@ -0,0 +1,6269 @@
...
+  small();
+
+  const smallCanvas = document.querySelector(".canvas-list label .c4");
+  smallCanvas.addEventListener('click', () => {
+    small();
+  });
+
+
+
+
```
Comment (karengyuzalyan) [inline script.js]: Redundant empty lines

File: script.js [hunk 6]
```diff
@@ -0,0 +1,6269 @@
...
+  var B = [
```
Comment (karengyuzalyan) [inline script.js]: Unclear name of variable, pleaase rename

Comment (karengyuzalyan) [reply script.js]: You can move this arrays to another file and import in this `script`, to have more readable code here

File: script.js [hunk 7]
```diff
@@ -0,0 +1,6269 @@
...
+      [
+        215,
+        49,
+        33,
+        183
+      ],
+      [
+        172,
+        49,
+        39,
```
Comment (karengyuzalyan) [inline script.js]: What does it mean `mid`, the name of funxtion should describe the action which do the function
for example - `calculateTheMiddle` or smth else

Comment (karengyuzalyan) [inline script.js]: Again recompose the name of the function

File: index.html [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+      </canvas>
+
+      <div class="canvas-list">
+        <ul>
+          <input id="cl-1" type="radio" name="canvas-list" value="" checked>
+          <label for="cl-1"><li class="c4">4x4</li></label>
+          <input id="cl-2" type="radio" name="canvas-list" value="">
+          <label for="cl-2"><li class="c32">32x32</li></label>
+          <input id="cl-3" type="radio" name="canvas-list" value="">
+          <label for="cl-3"><li class="c256">256x256</li></label>
```
Comment (karengyuzalyan) [inline index.html]: Please rename the class names `c256`, `c4` and `c32` to like `canvas-label__256x256`(and the others to appropriate way) or smth else

File: style.css [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  display: flex;
+  margin: 15px 18px;
+  max-width: 200px; }
+  .tools ul li:nth-child(2) {
+    margin-bottom: 30px; }
+  .tools ul li p {
+    margin-left: 35px; }
+.tools .border {
+  width: 100%;
+  border-bottom: 1px solid rgba(0, 0, 0, 0.12); }
```
Comment (karengyuzalyan) [inline style.css]: Please move `}` the closer bracket to new line, to make code more readable or just use some prettier plugin

File: index.html (hunk 1 above)
Comment (Elaray1) [reply index.html]: edited

File: index.html (hunk 2 above)
Comment (Elaray1) [reply index.html]: removed

File: script.js (hunk 3 above)
Comment (Elaray1) [reply script.js]: edited

File: script.js (hunk 4 above)
Comment (Elaray1) [reply script.js]: edited

File: script.js (hunk 5 above)
Comment (Elaray1) [reply script.js]: removed

File: script.js (hunk 6 above)
Comment (Elaray1) [reply script.js]: edited

File: script.js (hunk 7 above)
Comment (Elaray1) [reply script.js]: edited

Comment (Elaray1) [reply script.js]: edited

File: index.html (hunk 8 above)
Comment (Elaray1) [reply index.html]: edited

File: style.css (hunk 9 above)
Comment (Elaray1) [reply style.css]: edited

**Review by karengyuzalyan — APPROVED** [review]

**Pull request merged**